pkg/virtctl/imageupload/imageupload_test.go

```diff
@@ -700,6 +703,74 @@ var _ = Describe("ImageUpload", func() {
 			validatePVCDefaultInstancetypeLabels()
 		})
 
+		It("Should create DataSource pointing to the PVC", func() {
+			testInit(http.StatusOK)
+			cmd := clientcmd.NewRepeatableVirtctlCommand(
+				commandName, "dv", targetName,
+				"--size", dvSize,
+				"--uploadproxy-url", server.URL,
+				"--insecure",
+				"--force-bind",
+				"--datasource",
+				"--image-path", imagePath,
+			)
+			Expect(cmd()).To(Succeed())
+			Expect(pvcCreateCalled.IsTrue()).To(BeTrue())
+			Expect(kvtesting.FilterActions(&cdiClient.Fake, "create", "datasources")).To(HaveLen(1))
+
+			ds, err := cdiClient.CdiV1beta1().DataSources(targetNamespace).Get(context.Background(), targetName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(ds.Spec.Source.PVC).ToNot(BeNil())
+			Expect(ds.Spec.Source.PVC.Name).To(Equal(targetName))
+			Expect(ds.Spec.Source.PVC.Namespace).To(Equal(targetNamespace))
+			Expect(ds.Labels).ToNot(HaveKey(instancetypeapi.DefaultInstancetypeLabel))
+			Expect(ds.Labels).ToNot(HaveKey(instancetypeapi.DefaultInstancetypeKindLabel))
+			Expect(ds.Labels).ToNot(HaveKey(instancetypeapi.DefaultPreferenceLabel))
+			Expect(ds.Labels).ToNot(HaveKey(instancetypeapi.DefaultPreferenceKindLabel))
+		})
+
+		It("Should patch DataSource pointing to the PVC", func() {
+			testInit(http.StatusOK)
+
+			ds := &cdiv1.DataSource{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      targetName,
+					Namespace: targetNamespace,
+					Labels:    map[string]string{},
+				},
+				Spec: cdiv1.DataSourceSpec{
+					Source: cdiv1.DataSourceSource{
+						PVC: &cdiv1.DataVolumeSourcePVC{
+							Name:      targetName,
+							Namespace: targetNamespace,
+						},
+					},
+				},
+			}
+			_, err := cdiClient.CdiV1beta1().DataSources(targetNamespace).Create(context.Background(), ds, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			cmd := clientcmd.NewRepeatableVirtctlCommand(
+				commandName, "dv", targetName,
+				"--size", dvSize,
+				"--uploadproxy-url", server.URL,
+				"--insecure",
+				"--force-bind",
+				"--datasource",
+				"--image-path", imagePath,
+				"--default-preference", "centos",
+				"--default-preference-kind", "centos",
+			)
+			Expect(cmd()).To(Succeed())
+			Expect(pvcCreateCalled.IsTrue()).To(BeTrue())
+			Expect(kvtesting.FilterActions(&cdiClient.Fake, "patch", "datasources")).To(HaveLen(1))
+
+			ds, err = cdiClient.CdiV1beta1().DataSources(targetNamespace).Get(context.Background(), targetName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(ds.Labels).To(HaveKey(instancetypeapi.DefaultPreferenceLabel))
+			Expect(ds.Labels).To(HaveKey(instancetypeapi.DefaultPreferenceKindLabel))
```

Comment:
Expects for the instancetype and kind are missing.

Reply:
Can you please assert it updates the target PVC as well?

Reply:
> Expects for the instancetype and kind are missing.

Isn't only preference label enough, in order to display it in the UI?

Reply:
I've added instancetpye and kind and also added asserts of target PVC.